mlrun/db/httpdb.py

```diff
@@ -210,8 +217,14 @@ def api_call(
                     if isinstance(dict_[key], enum.Enum):
                         dict_[key] = dict_[key].value
 
-        if not self.session:
-            self.session = self._init_session()
+        retry_on_post = None
+        if method == "POST":
+            retry_on_post = self._is_retry_on_post_allowed(path)
+
+        # retry_on_post is not None means that method is POST, so we need to update the session
+        # with the appropriate retry policy
+        if not self.session or retry_on_post is not None:
+            self.session = self._init_session(retry_on_post or False)
```

Comment:
A slightly more readable suggestion
```suggestion
        # if the method is POST, we need to update the session with the appropriate retry policy
           if not self.session or method == "POST":
      retry = self._is_retry_on_post_allowed(path, method)
      self.session = self._init_session(retry)

def _is_retry_on_post_allowed(self, path: str, method):
   if method != "POST":
      return any(re.match(regex, path) for regex in self.RETRIABLE_POST_PATHS)
   return False
```